bspump/declarative/expression/datastructs/itemexpr.py

```diff
@@ -126,3 +129,43 @@ def __init__(self, orig, *, arg_with, arg_item, arg_default):
 
 	def __call__(self, context, event, *args, **kwargs):
 		return event.get(self.Key, self.Default)
+
+
+class ITEM_optimized_CONTEXT_VALUE(ITEM):
+
+	def __init__(self, orig, *, arg_with, arg_item, arg_default):
+		super().__init__(orig.App)
+
+		self.With = arg_with
+		self.Item = arg_item
+		if arg_default is None:
+			self.Default = arg_default
+		else:
+			# TODO: Default must be statically evaluated
+			raise NotImplementedError("")
+
+		self.Key = self.Item.Value
+
+		# TODO: Replace with JSON pointer path
+		if '.' in self.Key:
+			self.KeyList = self.Key.split('.')
+		else:
+			self.KeyList = None
+
+	def __call__(self, context, event, *args, **kwargs):
+
+		if self.KeyList is None:
+			return context.get(self.Key, self.Default)
```

Comment:
You don't want to have this `if` here ... which means that this needs to be two classes you specialises into.